pkg/cluster/config_test.go

```diff
@@ -1,244 +1,184 @@
-				NumWorkers:          defaults.NumWorkers,
-				KubeConfigFilePath:  filepath.Join(usr.HomeDir, ".kube", "kind-config-"+defaults.ClusterNameBase+strconv.Itoa(1)),
-				WaitForReady:        0,
-				NodeImageName:       "",
-				Retain:              false,
-				Tiller:              false,
-			}))
+	When("the CNI is weave", func() {
+		It("should set WaitForReady to 0", func() {
+			config := executePopulateConfig("", "weave", false)
+			Expect(config.Cni).To(Equal("weave"))
+			Expect(config.WaitForReady).To(Equal(time.Duration(0)))
 		})
+	})
 
-		It("Should create configs for 2 clusters with flannel and overlapping cidrs", func() {
-			usr, err := user.Current()
-			Ω(err).ShouldNot(HaveOccurred())
-
-			imageName := ""
-			cni := "flannel"
-			numClusters := 2
-
-			var clusters []*cluster.Config
-			for i := 1; i <= numClusters; i++ {
-				cl, err := cluster.PopulateConfig(i, imageName, cni, false, true, true, 5*time.Minute)
-				Ω(err).ShouldNot(HaveOccurred())
-				clusters = append(clusters, cl)
-			}
-
-			Expect(len(clusters)).Should(Equal(2))
-			Expect(clusters).Should(Equal([]*cluster.Config{
-				{
-					Cni:                 "flannel",
-					Name:                defaults.ClusterNameBase + strconv.Itoa(1),
-					PodSubnet:           "10.0.0.0/14",
-					ServiceSubnet:       "100.0.0.0/16",
-					DNSDomain:           defaults.ClusterNameBase + strconv.Itoa(1) + ".local",
-					KubeAdminAPIVersion: defaults.KubeAdminAPIVersion,
-					NumWorkers:          defaults.NumWorkers,
-					KubeConfigFilePath:  filepath.Join(usr.HomeDir, ".kube", "kind-config-"+defaults.ClusterNameBase+strconv.Itoa(1)),
-					WaitForReady:        0,
-					NodeImageName:       "",
-					Retain:              false,
-					Tiller:              true,
-				},
-				{
-					Cni:                 "flannel",
-					Name:                defaults.ClusterNameBase + strconv.Itoa(2),
-					PodSubnet:           "10.0.0.0/14",
-					ServiceSubnet:       "100.0.0.0/16",
-					DNSDomain:           defaults.ClusterNameBase + strconv.Itoa(2) + ".local",
-					KubeAdminAPIVersion: defaults.KubeAdminAPIVersion,
-					NumWorkers:          defaults.NumWorkers,
-					KubeConfigFilePath:  filepath.Join(usr.HomeDir, ".kube", "kind-config-"+defaults.ClusterNameBase+strconv.Itoa(2)),
-					WaitForReady:        0,
-					NodeImageName:       "",
-					Retain:              false,
-					Tiller:              true,
-				},
-			}))
+	When("the CNI is calico", func() {
+		It("should set WaitForReady to 0", func() {
+			config := executePopulateConfig("", "calico", false)
+			Expect(config.Cni).To(Equal("calico"))
+			Expect(config.WaitForReady).To(Equal(time.Duration(0)))
 		})
 	})
-})
+
+	When("the CNI is flannel", func() {
+		It("should set WaitForReady to 0", func() {
+			config := executePopulateConfig("", "flannel", false)
+			Expect(config.Cni).To(Equal("flannel"))
+			Expect(config.WaitForReady).To(Equal(time.Duration(0)))
+		})
+	})
+
+	When("overlapping IPs are desired", func() {
```

Comment:
Whilst the original test checked creating configuration of 2 clusters and verifying the subnets, this new one is more limited in scope and only tests a signle cluster, which leaves more room for error in the tested code.

Reply:
The original test didn't really add any value that I could see. It externally looped and called `PopulateConfig` but each call is isolated and wouldn't have any side effect to potentially cause the next call to fail.

Reply:
That said, I think this would be useful if it tested non-overlapping and verified different CIDRs across 2 clusters. I made that change with https://github.com/submariner-io/armada/pull/6/commits/16b0405e4f5f89ea440502bf997e05f2f737017a.

Reply:
Cool, that one's nice :)